File: Data Structures & Algorithms/two-integer-sum-ii/submission-0.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        left = 0
        right = 1
        while left < len(numbers):
            diff = target - numbers[left]
            if diff == numbers[right]:
                return [left + 1, right + 1]
            else:
                right += 1
            if right >= len(numbers):
                left += 1
                right = left + 1

File: Data Structures & Algorithms/palindromic-substrings/submission-0.java
class Solution {
    public int countSubstrings(String s) {
        int result = 0;

        for (int i = 0; i < s.length(); i++) {
            int countOdd = find(s, i, i);
            int countEven = find(s, i, i + 1);

            result += (countOdd + countEven);

        }

        return result;
    }

    private int find(String s, int left, int right) {
        int count = 0;

        while (left >= 0 && right < s.length() && s.charAt(left) == s.charAt(right)) {
            left--;
            right++;
            count++;
        }

        return count;
    }
}
